tools/cmake/mbed_greentea.cmake

```diff
@@ -1,12 +1,19 @@
 # Copyright (c) 2020-2021 ARM Limited. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-option(MBED_TEST_BAREMETAL OFF)
+option(MBED_GREENTEA_TEST_BAREMETAL "Select baremetal greentea tests" OFF)
 
-set(MBED_CONFIG_PATH ${CMAKE_CURRENT_BINARY_DIR} CACHE INTERNAL "")
+set(MBED_TARGET_SERIAL_PORT "" CACHE STRING "Serial port of the DUT")
+set(MBED_TARGET_SERIAL_NUM "" CACHE STRING "USB Serial number of the DUT")
+set(MBED_TARGET_MOUNT_POINT "" CACHE STRING "Mount point for the DUT")
```

Comment:
It's a limitation of CTest that we can't pass any options at run time 😢: https://gitlab.kitware.com/cmake/cmake/-/issues/20470
If a user switches to a different board or connects a board after compilation, they need to rerun cmake to use their device. Though this won't require recompilation.

Reply:
Correct. Users will need to run the CMake config and generate step again if they want to change the cache values.